File: 0069-sqrtx/0069-sqrtx.cpp
class Solution {
public:
    int mySqrt(int x) {
        long long int left=0,right=100000;
        long long int mid=(left+right)/2;
        while(true)
        {
            long long int temp=mid;
            if (mid*mid>x) right=mid+1;else left=mid;
            mid=(left+right)/2;
            if (temp==mid) break;
        }
            
        return left;
    }
};

File: 0069-sqrtx/0069-sqrtx.py
class Solution:
    def mySqrt(self, x: int) -> int:
        if x ==0 :
            return 0
        ans = 1
        low,high = 1, x
        while low <= high:
            mid = (low + high) // 2
            if mid * mid <= x:
                ans = mid
                low += 1
            else:
                high = mid - 1
        return ans